feat: Improving Contextual Understanding with contentScope Configuration

syuzanvs opened the pull request:
- Introduced `contentScope` property in `augmentation.ts` for enhanced editor module declarations.
- Added `contentScope` attribute in `PromptHelper` class to allow specifying a DOM scope for context extraction.
- Updated the `getSystemPrompt` method to dynamically retrieve content from elements matching the `contentScope` selector.
- Ensured compatibility with existing functionality by defaulting to the previous behavior when `contentScope` is undefined.

This enhancement improves flexibility in handling editor contexts, enabling more precise content retrieval in scoped environments.

Closes #86

jjroelofs reviewed (changes requested):
Current logic processes all matching elements, while the requirement is to match only the first element upwards from the active editor. Use `closest()` for this.
We do this because selectors like `form` or `.az-container` can match multiple elements of unrelated forms or page-builder containers. This is why we only match the first ancestor that matches the give selector.

**Example improved code (generated, not tested):**
```typescript
if (this.contentScope) {
    const activeEditorElement = this.editor.editing.view.getDomRoot();
    const targetElement = activeEditorElement?.closest(this.contentScope);
    context = targetElement?.innerHTML ?? '';
}
```

#### **Documentation Improvement**
Clarify in `README.md` that `contentScope` uses a CSS selector to find the closest matching ancestor of the active editor. Example:
```md
`contentScope`: CSS selector for determining the scope of content processing. Matches the closest ancestor element to the active editor.
```

#### **Best Practices**
- Use CKEditor’s `DomConverter` to ensure consistency when accessing the DOM.
- Optimize the README to specify that only one element is processed, avoiding concatenation confusion.

This refinement ensures the patch meets its stated goals with clarity and efficiency.

syuzanvs committed:
feat: Improving Contextual Understanding with contentScope Configuration

- Introduced `contentScope` property in `augmentation.ts` for enhanced editor module declarations.
- Added `contentScope` attribute in `PromptHelper` class to allow specifying a DOM scope for context extraction.
- Updated the `getSystemPrompt` method to dynamically retrieve content from elements matching the `contentScope` selector.
- Ensured compatibility with existing functionality by defaulting to the previous behavior when `contentScope` is undefined.

This enhancement improves flexibility in handling editor contexts, enabling more precise content retrieval in scoped environments.

Closes #86

jjroelofs reviewed (approved):
great
